app/uk/gov/hmrc/lisaapi/config/AppContext.scala

```diff
@@ -21,22 +21,22 @@ import play.api.Configuration
-  lazy val appUrl = runModeConfiguration.getString("appUrl").getOrElse(throw new RuntimeException("appUrl is not configured"))
+class AppContext @Inject()(config: Configuration, runMode: RunMode) extends ServicesConfig(config, runMode) {
+  lazy val appName: String = config.get[String]("appName")
+  lazy val appUrl: String = config.get[String]("appUrl")
   lazy val serviceLocatorUrl: String = baseUrl("service-locator")
-  lazy val registrationEnabled: Boolean = runModeConfiguration.getBoolean(s"microservice.services.service-locator.enabled").getOrElse(false)
-  lazy val apiContext = runModeConfiguration.getString("api.context").getOrElse(throw new RuntimeException(s"Missing Key api.context"))
-  lazy val baseUrl = runModeConfiguration.getString(s"baseUrl").getOrElse(throw new RuntimeException(s"Missing Key baseUrl"))
-  lazy val v1apiStatus = runModeConfiguration.getString("api.status").getOrElse(throw new RuntimeException(s"Missing Key api.status"))
-  lazy val v2apiStatus = runModeConfiguration.getString("api.statusv2").getOrElse(throw new RuntimeException(s"Missing Key api.statusv2"))
-  lazy val desAuthToken = runModeConfiguration.getString("desauthtoken").getOrElse(throw new RuntimeException(s"Missing Key desauthtoken"))
-  lazy val desUrlHeaderEnv: String =  runModeConfiguration.getString("environment").getOrElse(throw new RuntimeException(s"Missing Key environment"))
-  lazy val access = runModeConfiguration.getConfig(s"api.access")
-  lazy val v1endpointsEnabled = runModeConfiguration.getBoolean("api.endpointsEnabled").getOrElse(throw new RuntimeException(s"Missing key api.endpointsEnabled"))
-  lazy val v2endpointsEnabled = runModeConfiguration.getBoolean("api.endpointsEnabledv2").getOrElse(throw new RuntimeException(s"Missing key api.endpointsEnabledv2"))
+  lazy val registrationEnabled: Boolean = config.getOptional[Boolean](s"microservice.services.service-locator.enabled").getOrElse(false)
+  lazy val apiContext: String = config.get[String]("api.context")
+  lazy val baseUrl: String = config.get[String](s"baseUrl")
+  lazy val v1apiStatus: String = config.get[String]("api.status")
+  lazy val v2apiStatus: String = config.get[String]("api.statusv2")
+  lazy val desAuthToken: String = config.get[String]("desauthtoken")
+  lazy val desUrlHeaderEnv: String =  config.get[String]("environment")
+  lazy val access: Option[Configuration] = config.getOptional[Configuration](s"api.access")
+  lazy val v1endpointsEnabled: Boolean = config.get[Boolean]("api.endpointsEnabled")
+  lazy val v2endpointsEnabled: Boolean = config.get[Boolean]("api.endpointsEnabledv2")
 
   def endpointIsDisabled(endpoint: String): Boolean = {
-    runModeConfiguration.getStringList("api.disabledEndpoints").fold(false)(list => list.contains(endpoint))
+    config.getOptional[Seq[String]]("api.disabledEndpoints").fold(false)(list => list.contains(endpoint))
```

Comment:
Did you manually validate this works? I know the test mocks out the call. I might be wrong but I think I've had an issue when pulling in lists before and had to base64 the value and bring it it.  It might not be the same as this instance but just wanted to mention it.

Reply:
tested it by running the api tests after changing the config to disable one of the endpoints and tests failed for correct reason when checking the logs